package/MDAnalysis/coordinates/memory.py

```diff
@@ -264,6 +264,14 @@ def __init__(self, coordinate_array, order='fac',
         super(MemoryReader, self).__init__()
         self.filename = filename
         self.stored_order = order
+
+        try:
+            if len(coordinate_array.shape) == 2 and coordinate_array.shape[1] == 3:
+                    coordinate_array = coordinate_array[np.newaxis, :, :]
+        except AttributeError as e:
+            raise ValueError("Input to the MemoryReader is of improper format."
+                )
+
         self.set_array(np.asarray(coordinate_array), order)
```

Comment:
Remove `np.asarray(coordinate_array)` – we are now **requiring numpy arrays** and your check above raises `TypeError` if it is not the right type.

Reply:
- Update docs with a *Raises* section
- Update docs with stating that the `MemoryReader` now requires ndarray
- Add a `versionchanged:: 0.17.1` section that states that input now hast to be a ndarray